framework/include/problems/FEProblemBase.h

```diff
@@ -2736,6 +2736,9 @@ class FEProblemBase : public SubProblem, public Restartable
 
   std::vector<SolverParams> _solver_params;
 
+  /// Default blocks for block restriction (kernel and material coverage check only for the default blocks)
```

Comment:
I think this doxygen is misleading. This data member is really only about coverage checks right? It doesn't really have anything to do with default block restriction. I think it's important to get this doxygen right because in theory `FEProblemBase` could very well set block restriction to `_blocks` when object `InputParameters` pass through the problem via the `addXXX` methods

Reply:
I fixed the docstring.

Let me think about the second part of your suggestion. If it's not hard to do I may as well try that now.